src/lowering.rs

```diff
@@ -63,6 +70,25 @@ impl InitVisitor {
         self.ctxt.scope(old);
     }
 
+    fn collect_user_init_candidates(&mut self, unit: &CompilationUnit) {
+        // collect all candidates for user-defined init functions
+        for pou in unit.pous.iter().filter(|it| matches!(it.kind, PouType::FunctionBlock | PouType::Program))
+        {
+            // add the POU to potential `FB_INIT` candidates
+            self.user_inits
+                .insert(pou.name.to_owned(), self.index.find_method(&pou.name, "FB_INIT").is_some());
+        }
+
+        for user_type in
+            unit.user_types.iter().filter(|it| matches!(it.data_type, DataType::StructType { .. }))
+        {
+            // add the struct to potential `STRUCT_INIT` candidates
+            if let Some(name) = user_type.data_type.get_name() {
+                self.user_inits.insert(name.to_string(), false);
```

Comment:
@ghaith If we want to support user-defined struct-init functions for FFI purposes, their addition should be trivial if we use a naming-scheme here. We can then just query the index if such a function is present and then add it to the hashmap to be generated/called.